modules/c++/re/source/RegexPCRE.cpp

```diff
@@ -114,6 +114,15 @@ class ScopedMatchData
         const size_t index = outVector[idx * 2];
         const size_t end = outVector[idx * 2 + 1];
 
+        // If index is equal to end then we need to return an empty string.
+        // It is not always the case when this happens that these values will
+        // be less than or equal to the size of the string, so we need to
+        // explicitly return "" so nothing goes out of bounds.
+        if (index == end)
```

Comment:
Fixed.

Note, there are other situations where index == end and they do not equal PCRE2_UNSET. From what I can see, they are all correctly handled with the substr command though and return an empty string.